stringslab.js

```diff
@@ -0,0 +1,158 @@
+// # String Exercises
+//
+// 1. `console.log` the first character of a string.
+let quote = 'A friend is one that knows you as you are, \nunderstands where you have been, \naccepts what you have become, \nand still, gently allows you to grow'
+
+console.log(quote[0]);
+
+// 2. `console.log` the length of a string.
+console.log(quote.length);
+//
+// 3. `console.log` the last character of any string.
+console.log(quote[quote.length - 1]);
+
+
+// 4. Create a code block that takes a single string variable,
+// and returns a copy of the string with the last letter
+// capitalized. For example:
+//
+// ```js
+// 'jimmy'
+// // => 'jimmY'
+// ```
+
+let newQuote = quote.slice(0, quote.length - 1) + quote[quote.length - 1].toUpperCase()
+console.log(newQuote)
+
+
+// 5. Create a drEvil code block that will take a single
+//number variable, and log the '<variableAmount> dollars',
+// your code block should add '(pinky)' at the end of the
+ //amount if it's 1 million. For example:
+//
+
+let x = 10
+console.log(`$ ${x} dollars`);
+
+let y = 1000000
+console.log(`$ ${y} dollars (pinky)`);
+
+
+// ```js
+// let amount = 10
+// // => 10 dollars
+// let amount = 1000000
+// // =>  1000000 dollars (pinky)
+// ```
+//
+// 6. Create a `verbing` code block. It should take a single
+//string variable. If its length is at least 3, it should
+//add 'ing' to its end, unless it already ends in 'ing',
+//in which case it should add 'ly' instead. If the
+//string length is less than 3, it should leave it unchanged.
+// For example:
+//
+// ```js
+// verbing('box')
+// // => 'boxing'
+// verbing('train')
+// // => 'training'
+// verbing('swimming')
+// // =>  'swimmingly'
+// verbing('go')
+// // =>  'go'
+// ```
+//
+let verb = `disenchanting`
+verb.endsWith('ing')
+if (verb.length >= 3 && !verb.endsWith('ing')) {
+  console.log(`${verb} ing`)
+} else if (verb.endsWith('ing')) {
+  console.log(`${verb}ly`)
+} else {
+  console.log(verb)
+}
+
+// 7. Create a `withoutLast` code block that takes a single string
+// variable, and returns a copy of the string without the last word.
+//
+let sameQuote = 'A friend is one that knows you as you are, \nunderstands where you have been, \naccepts what you have become, \nand still, gently allows you to grow'
+let withoutLast = sameQuote.slice(0, 140)
```

Comment:
This is hard coding as slice(0, 140) would not work if we change sameQuote to “All the things I really like to do are either immoral, illegal or fattening.”. Always try to avoid hard coding as it is bad practice as a developer. Test your code with different inputs before moving on to next problem.